docs/devguide/debugging_guide.rst

```diff
@@ -0,0 +1,70 @@
+
+The diagram is made with `WebSequenceDiagrams`_.
+
+To see if the DHCP request was received by ISC DHCP, look in `/var/log/syslog` of the RackHD host. `grep DHCP /var/log/syslog` works reasonably well - you're looking for a sequence like this::
+
+    Jan  8 15:43:43 rackhd-demo dhclient: DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 3 (xid=0x5b3b9260)
+    Jan  8 15:43:43 rackhd-demo dhclient: DHCPREQUEST of 10.0.2.15 on eth0 to 255.255.255.255 port 67 (xid=0x60923b5b)
+    Jan  8 15:43:43 rackhd-demo dhclient: DHCPOFFER of 10.0.2.15 from 10.0.2.2
+    Jan  8 15:43:43 rackhd-demo dhclient: DHCPACK of 10.0.2.15 from 10.0.2.2
+
+You should also see the DHCP proxy return the bootfile. In the DHCP-proxy logs, look for lines with `DHCP.messageHandler`::
+
+    S 2016-01-08T19:31:43.268Z [on-dhcp-proxy] [DHCP.messageHandler] [Server] Unknown node 08:00:27:f3:9f:2e. Sending down default bootfile.
+
+And immediately thereafter, you should see the server request the file from TFTP::
+
+    S 2016-01-08T19:31:43.352Z [on-tftp] [Tftp.Server] [Server] tftp: 67.300 monorail.ipxe
+
+.. _WebSequenceDiagrams: https://www.websequencediagrams.com
+
+
+Default discovery workflow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code::
+
+    title Default Discovery Workflow
+    Server->RackHD: DHCP from PXE(nic or BIOS)
+    RackHD->Server: ISC DHCP response with IP
+    RackHD->Server: DHCP-proxy response with bootfile
+    note right of RackHD: If the node is already "known", it will only respond if there's an active workflow that's been invoked related to the node
+    Server->RackHD: Request to download bootfile via TFTP
+    RackHD->Server: TFTP sends requested file (monorail.ipxe)
+    note left of Server: Server loads monorail.ipxe and initiates on bootloader
+    Server->RackHD: IPXE script requests what to do from RackHD (http)
+    note right of RackHD: RackHD looks up IP address of HTTP request from IPXE script to find the node via it's mac-address.
+    note right of RackHD: If the node is already "known", it will only respond if there's an active workflow that's been invoked related to the node
+    note right of RackHD: If the node isn't known, it will create a workflow (default is a discovery workflow) and respond with an iPXE script to initiate that
```

Comment:
Strictly speaking, it should be "SKU discovery workflow" (Graph.Sku.Discovery, https://github.com/RackHD/on-http/blob/master/lib%2Fservices%2Fprofiles-api-service.js#L123), the discovery workflow (Graph.Discovery) is nested in Graph.Sku.Discovery.